core/arch/aarch64/instr_create.h

```diff
@@ -311,7 +311,10 @@
  * \param d  The opnd_t explicit destination operand for the instruction.
  * \param s  The opnd_t explicit source operand for the instruction.
  */
-#define XINST_CREATE_add_s(dc, d, s) INSTR_CREATE_adds(dc, d, d, s)
+#define XINST_CREATE_add_s(dc, d, s) \
+  opnd_is_reg(s) ? \
```

Comment:
general macro safety: Include containing parens to isolate the ?: from whatever it's embedded into